hooks/playbooks/barbican-enable-luna.yml

```diff
@@ -0,0 +1,89 @@
+        barbican_dest_image_namespace: "{{ cifmw_set_openstack_containers_namespace }}"
+        barbican_dest_image_tag: "{{ cifmw_update_extras['cifmw_set_openstack_containers_tag'] }}{{ cifmw_update_barbican_custom_tag }}"
+        image_registry_verify_tls: "{{ cifmw_image_registry_verify_tls | default('false', true) }}"
+        luna_minclient_src: "{{ cifmw_hsm_luna_minclient_src }}"
+
+    - name: Create secrets with the HSM certs and hsm-login credentials
+      ansible.builtin.include_role:
+        name: rhoso_luna_hsm
+        tasks_from: create_secrets
+      vars:
+        luna_client_name: "{{ cifmw_hsm_client_name }}"
+        chrystoki_conf_src: "{{ cifmw_hsm_chrystoki_conf_src }}"
+        luna_server_cert_src: "{{ cifmw_hsm_luna_server_cert_src }}"
+        luna_client_cert_src: "{{ cifmw_hsm_luna_client_cert_src }}"
+        luna_partition_password: "{{ cifmw_hsm_partition_password }}"
+        kubeconfig_path: "{{ cifmw_openshift_kubeconfig }}"
+        oc_dir: "{{ cifmw_path }}"
+        luna_data_secret: "{{ cifmw_hsm_luna_client_data_secret | default('barbican-luna-client-data', true) }}"
+        login_secret: "{{ cifmw_hsm_login_secret | default('barbican-luna-login', true) }}"
+
+- name: Create kustomization to use update barbican to use luna
+  hosts: "{{ cifmw_target_hook_host | default('localhost') }}"
+  tasks:
+    - name: Create file to customize barbican resource deployed in the control plane
+      vars:
+        client_data_secret: "{{ cifmw_hsm_luna_client_data_secret | default('barbican-luna-client-data', true) }}"
+        login_secret: "{{ cifmw_hsm_login_secret | default('barbican-luna-login', true) }}"
+      ansible.builtin.copy:
+        dest: "{{ cifmw_basedir }}/artifacts/manifests/kustomizations/controlplane/93-barbican-luna.yaml"
+        content: |-
+          apiVersion: kustomize.config.k8s.io/v1beta1
+          kind: Kustomization
+          resources:
+          namespace: {{ namespace }}
+          patches:
+          - target:
+              kind: OpenStackControlPlane
+              name: .*
+            patch: |-
+              - op: add
+                path: /spec/barbican/template/globalDefaultSecretStore
+                value: pkcs11
+              - op: add
+                path: /spec/barbican/template/enabledSecretStores
+                value:
+                  - pkcs11
+              - op: add
+                path: /spec/barbican/template/pkcs11
+                value:
+                  loginSecret: {{ login_secret }}
+                  clientDataSecret: {{ client_data_secret }}
+                  clientDataPath: /usr/local/luna/config
+              - op: add
+                path: /spec/barbican/template/customServiceConfig
+                value: |
+                  [p11_crypto_plugin]
+                  plugin_name = PKCS11
+                  library_path = /usr/local/luna/libs/64/libCryptoki2.so
```

Comment:
This probably needs to be configurable.  With newer Luna Minimal Clients, the path here should be `/usr/local/luna/libs/64/libCryptoki2_64.so`